llvm/lib/Target/AIE/AIECombinerHelper.cpp

```diff
@@ -901,6 +916,31 @@ bool modifyToS20(InstrNode Start, MachineRegisterInfo &MRI, MachineIRBuilder &B,
     Helper.tryCombineCopy(*StartNodeMI);
     return true;
   }
+  case TargetOpcode::G_INTRINSIC: {
+    if (isS20Intrinsic(*StartNodeMI, MRI)) {
```

Comment:
I have a feeling that we need to refactor some subtarget-independent stuff in this file to TII. I think we can use and approach like `getVExtractOpInfo` in this same file. My goal is to generalize this CombinerHelper as much as possible.

Reply:
+1.  
Luckily we do not have many hardcoded aie2_* intrinsic in the file.